### PR TITLE
Added skeleton to access chart data using servlet to pass to frontend.

### PR DESCRIPTION
**Pull request by Jaxobc**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ChartServlet.Java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@WebServlet("/chart-data") public class ChartServlet extends HttpServlet {
+  @Override public void doGet(HttpServletRequest request,
+                              HttpServletResponse response) throws IOException {
+    List<DataPair> dataPairs = new ArrayList<DataPair>();
+
+    Gson gson = new Gson();
+
+    // Send the JSON as the response
+    response.setContentType("application/json");
+    response.getWriter().println(gson.toJson(dataPairs));
```
Comment (kofimeighan) [inline portfolio/src/main/java/com/google/sps/servlets/ChartServlet.Java]: nit: you can convert to json using 1 line
>new Gson().toJson(dataPairs)

File: portfolio/src/main/webapp/js/custom-script.js [hunk 2]
```diff
@@ -275,3 +277,15 @@ function drawStateIncarcerationChart() {
       document.getElementById('chart-container'));
   chart.draw(stateData, stateChartDimensions);
 }
+
+async function loadChartData() {
+  const response = await fetch('/data');
+  const dataPairs = await response.json();
```
Comment (kofimeighan) [inline portfolio/src/main/webapp/js/custom-script.js]: should this data point be /charts-data?

File: portfolio/src/main/java/com/google/sps/servlets/ChartServlet.Java (hunk 1 above)
Comment (Jaxobc) [reply portfolio/src/main/java/com/google/sps/servlets/ChartServlet.Java]: That makes sense.

File: portfolio/src/main/webapp/js/custom-script.js (hunk 2 above)
Comment (Jaxobc) [reply portfolio/src/main/webapp/js/custom-script.js]: Yes it should be.

**Review by kennzoid — APPROVED** [review]

**Review by briafassler9 — APPROVED** [review]

**Pull request merged**